lib/atlas/dataset/derived_dataset.rb

```diff
@@ -0,0 +1,27 @@
+module Atlas
+  class Dataset::DerivedDataset < Dataset
+    GRAPH     = 'graph.yml'.freeze
+
+    attribute :base_dataset,      String
+    attribute :scaling,           Hash
+
+    validate :base_dataset_exists
+
+    def graph
+      YAML.load_file(File.join(directory, area, GRAPH))
+    end
+
+    # Overwrite
+    def dataset_dir
+      @dataset_dir ||= Atlas.data_dir.join(DIRECTORY).join(Dataset::FullDataset.find(base_dataset).key.to_s)
+    end
+
+    #######
+    private
+    #######
+
+    def base_dataset_exists
```

Comment:
Nitpicking, but I think this exceeds 80 characters (I'm guessing `dataset_dir` also does). The `exists?` call [should have parentheses around it](https://github.com/bbatsov/ruby-style-guide#method-invocation-parens):

```ruby
    # ...

    def base_dataset_exists
      unless Dataset::FullDataset.exists?(base_dataset)
        errors.add(:base_dataset, 'does not exist')
      end
    end

    # ...
```